src/components/core/Text/Text.js

```diff
@@ -1,29 +1,26 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { text } from '../../../mixins';
 
-const TextBase = styled.p`
-	margin: 0;
-	font-family: ${props => props.theme.fontFamily.base};
-	font-size: ${props =>
-		props.theme.fontSize[props.tertiary ? 'small' : 'base']};
-	font-weight: normal;
-	color: ${props =>
-		props.theme.color[
-			(props.secondary && 'secondary') ||
-				(props.tertiary && 'secondary') ||
-				(props.error && 'error') ||
-				'base'
-		]};
-`;
+const Base = ({ is: Component, ...props }) => <Component {...props} />;
 
 /**
  * A component to render all text in the app.
  */
-const Text = ({ children, is, ...props }) => {
-	const Tag = TextBase.withComponent(is);
-	return <Tag {...props}>{children}</Tag>;
-};
+const Text = styled(Base)`
+	margin: 0;
+	${props =>
+		text({
```

Comment:
I’m really not liking mixing thing in one. This feels too coupled and not really reusable to me.

Reply:
Yeah, I guess `fontSize({fontSize: 'base'}); color({color: 'base'})...` would be enough, we don't need it that often, and it's only for primitive components.